Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -151,6 +153,44 @@ private async Task ProcessResult(Split split, CancellationToken cancellationToke
       callback.OnProgress?.Invoke("VCprove", splitNumber < 0 ? 0 : splitNumber, total, provenCost / (remainingCost + provenCost));
 
       if (!proverFailed) {
+        if (callback is VerificationResultCollector collector) {
+          List<AssertCmd> asserts = split.AssertCmds.ToList();
+          Dictionary<AssertCmd, Outcome> perAssertOutcome = new();
+          Dictionary<AssertCmd, Counterexample> perAssertCounterExamples = new();
+          if (outcome == Outcome.Correct) {
+            perAssertOutcome = asserts.ToDictionary(cmd => cmd, assertCmd => Outcome.Correct);
+          } else {
+            foreach (var counterExample in collector.examples) {
```

Comment:
Why does each Split need to look at counterExamples collected by other splits?

I would think that each split reports a CounterExample if it failed, and the CounterExample can say which statements where involved in the failure, which we can then mark red in the IDE if that CounterExample is active.

Btw, I saw that in the IDE you also mark asserts green. Do you mark them green if the assertion passed for each split in which it occurred ? What do you do if an assertion passes for the first completed split, but fails for the split that completes after it?

Reply:
This code is invoked after all the counter-examples of that split have been emitted. However, counter-examples are not recorded per split result (not even in the current master), so I have to rely on the entire list of previously emitted counter-examples, and disregarding counter-examples that were emitted by other splits. Let me know how I can do better, e.g. maybe store the counter-examples in VCResult?
Each split reports up to 5 counter examples.

To my understanding, I think that splits are all disjoints, so if an assert happens in one split, it shouldn't appear in another split. Yet I don't rely on that anyway. If a single assert would appear in two different splits, they will be simply be recorded as two different assertions at the same place. Hence, if one of them fail, the error will be given priority on the gutter - the gutter rendering is like a z-buffer where errors decorations are "closer" than verified decorations.

Reply:
I think it may make sense, as you're beginning to suggest here, to adapt `VCResult` to include counterexamples and multiple outcomes (using the logic you implement in this PR). That will require a lot less work from clients of that type in most cases, I expect.